src/posix/platform/ip6_utils.hpp

```diff
@@ -45,13 +45,10 @@ class Ip6AddressString
     /**
      * The constructor of this converter.
      *
-     * @param[in]   aAddress    A pointer to a buffer holding an IPv6 address.
+     * @param[in]   aAddress    A pointer to an IPv6 address (MUST NOT be nullptr).
      *
      */
-    Ip6AddressString(const void *aAddress)
-    {
-        VerifyOrDie(inet_ntop(AF_INET6, aAddress, mBuffer, sizeof(mBuffer)) != nullptr, OT_EXIT_ERROR_ERRNO);
-    }
+    Ip6AddressString(const otIp6Address *aAddress) { otIp6AddressToString(aAddress, mBuffer, sizeof(mBuffer)); }
```

Comment:
intended to use void* to accept both otIp6Address and POSIX IPv6 address structures. Note inet_ntop's prototype is void*. I prefer keeping it as is. Thoughts?

Reply:
We don't seem to use `void *` currently from anywhere, but I don't think it matters.  Will change it back to `void *`

Reply:
Actually thinking more on this, I think it's better to keep the posix as it was,
- The `inet_ntop()` is available on posix and also it can give shorter (optimized/compressed format) with `::` .
- We don't really care about code size on posix platform, so little reason to use OT API. 

I will change the `ip6_utils.hpp` back to what it was.